### PR TITLE
Tracker update states

### PR DESCRIPTION
**Pull request by cjb**
@keybase/react-hackers

I wanted to add some new trackerStates that @gabriel and I can each start using in trackerv2 straight away.  "followed" and "refollowed" correspond to states where you've performed some action and now the tracker popup just needs to say "Close".  There's also a "followedProofsAdded" state -- for when someone you follow added new proofs -- which we won't use immediately because we don't plan to do a popup for that yet.

The additions to trackerv1's `statusColor` map are necessary because it (and maybe other parts of the code!) makes the mistake of thinking that `SimpleProofState` is a *per-proof* state that it should make per-proof rendering decisions based on.  But `SimpleProofState` is actually the same thing as the "global" trackerState -- we probably should've called it `SimpleTrackerState` instead.

I think that the `revoked` entry in `SimpleProofState` is actually a per-proof state that snuck in to our global tracker states.  But I guess I can re-use it for `unfollowed` without harm?

### CODE REVIEW

File: shared/constants/tracker.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /* @flow */
 
 // Simple state of the overall proof result
-export type SimpleProofState = 'normal' | 'warning' | 'error' | 'checking' | 'revoked'
+export type SimpleProofState = 'normal' | 'warning' | 'error' | 'checking' | 'revoked' | 'followed' | 'refollowed' | 'followedProofsAdded'
```
Comment (gabriel) [inline shared/constants/tracker.js]: Can you add a comment which explains followedProofsAdded...

**Comment by gabriel** [comment]
export type SimpleProofState = 'normal' | 'warning' | 'error' | 'checking' | 'revoked' | 'followed' | 'refollowed' | 'followedProofsAdded'

does normal mean not following? Maybe rename that (or add a TODO to rename after we nuke v1)

also after nuking v1 we should take out checking and make it a separate bool property..

e.g. its nice to know if you are following someone while it's checking

**Comment by cjb** [comment]
Judging by convo on Slack, we're actually gonna split out TrackerState instead.

**Comment by cjb** [comment]
@gabriel How about:

export type TrackerState = 'following' | 'warning' | 'error' | 'checking' | 'unfollowed' | 'followed' | 'refollowed' | 'newProofs'

**Comment by gabriel** [comment]
Sure...

Should we take out warning and error and make them part of props?

Maybe you are following + warning? or there was an error trying to refollow?
Presence of error means there is an error?

state: TrackerState
warning: Warning
error: Error // Error to show (supercedes any warnings, if also present)

Also is it good for enums to have a "blank" state? like 'none' or 'unknown'?

**Comment by gabriel** [comment]
btw, am ok with these changes if you want to merge... we can refactor this stuff more easily once v1 is nuked also

**Comment by cjb** [comment]
@gabriel Actually totally changed my mind, sorry -- the SimpleProofState is computed from the full state, it doesn't really make sense to inject new states into it, I think I should just fire actions for followed/unfollowed/refollowed that set props on the components.

That doesn't tell us what to do about newProofs.  It works fine in practice, we just look at the meta state.  But it's hard for you to mock on the components page.

**Comment by gabriel** [comment]
ok sure

**Comment by cjb** [comment]
Closing in favor of newer PR.

**Pull request closed without merging**